### ISSUE TITLE
Upgrade to Styled Components v5

### ISSUE THREAD

**Issue by simonsinclair** [bbc/simorgh#5182]
**Is your feature request related to a problem? Please describe.**
Upgrade to the latest release of Styled Components v5. It is non-breaking and compared to v4, it comes with:
- 26% smaller bundle size (16.2kB vs. 12.42kB min+gzip)
- 22% faster client-side mounting 
- 26% faster updating of dynamic styles 
- 45% faster server-side rendering 

**Describe the solution you'd like**

Simorgh:
- Update `styled-components` to the latest version.
- Update `jest-styled-components` to the latest version.
- Check for regressions.

Psammead:
- Update `styled-components` to the latest version.
- Update `jest-styled-components` to the latest version.
- Update `psammead-test-helpers` to use the latest `jest-styled-components`.
- Check for regressions.

**Describe alternatives you've considered**
n/a

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**
- https://medium.com/styled-components/announcing-styled-components-v5-beast-mode-389747abd987

**Comment by joshcoventry** [comment]
Blocked by https://github.com/bbc/psammead/issues/3054

**Comment by simonsinclair** [comment]
Closing in favour of migrating to Emotion, which is tracked in the following issue: https://github.com/bbc/simorgh/issues/7772.
